From dfd3e2cf4d7eebbd96bc6d4ffcb89668b0128ad6 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Wed, 14 Dec 2016 19:35:34 +0800
Subject: [PATCH 1356/5242] MLK-13610-1 ARM: dts: add SD3.0 support for usdhc0
 on imx7ulp-evk

commit  d32a4761b7b21237dac5eaf6bcf7c40c3b47989d from
https://source.codeaurora.org/external/imx/linux-imx.git

Add SD3.0 support for usdhc0 on imx7ulp-evk board. Currently the
usdhc0 root clock is 158.4MHz.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-evk.dts |   23 +++++++++++++++++++----
 arch/arm/boot/dts/imx7ulp.dtsi    |    2 ++
 2 files changed, 21 insertions(+), 4 deletions(-)

diff --git a/arch/arm/boot/dts/imx7ulp-evk.dts b/arch/arm/boot/dts/imx7ulp-evk.dts
index 4dce316..fa74188 100644
--- a/arch/arm/boot/dts/imx7ulp-evk.dts
+++ b/arch/arm/boot/dts/imx7ulp-evk.dts
@@ -57,6 +57,16 @@
 			gpio = <&gpio0 0 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_vsd_3v3: regulator@1 {
+			compatible = "regulator-fixed";
+			reg = <1>;
+			regulator-name = "VSD_3V3";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&gpio1 0 GPIO_ACTIVE_HIGH>;
+			enable-active-high;
+		};
 	};
 
 	extcon_usb1: extcon_usb1 {
@@ -130,8 +140,9 @@
 	imx7ulp-evk {
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
-				IMX7ULP_PAD_PTC10__PTC10	0x30100
+				IMX7ULP_PAD_PTC10__PTC10		0x30100		/* USDHC0 CD */
 				IMX7ULP_PAD_PTC1__PTC1		0x20100
+				IMX7ULP_PAD_PTD0__PTD0		0x30100		/* USDHC0 RST */
 			>;
 		};
 
@@ -171,7 +182,7 @@
 		pinctrl_usdhc0: usdhc0grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTD1__SDHC0_CMD	0x843
-				IMX7ULP_PAD_PTD2__SDHC0_CLK	0x843
+				IMX7ULP_PAD_PTD2__SDHC0_CLK	0x10843
 				IMX7ULP_PAD_PTD7__SDHC0_D3	0x843
 				IMX7ULP_PAD_PTD8__SDHC0_D2	0x843
 				IMX7ULP_PAD_PTD9__SDHC0_D1	0x843
@@ -347,10 +358,14 @@
 };
 
 &usdhc0 {
-	pinctrl-names = "default", "sleep";
+	pinctrl-names = "default", "state_100mhz", "state_200mhz", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc0>;
 	pinctrl-1 = <&pinctrl_usdhc0>;
-	non-removable;
+	pinctrl-2 = <&pinctrl_usdhc0>;
+	pinctrl-3 = <&pinctrl_usdhc0>;
+	cd-gpios = <&gpio0 10 GPIO_ACTIVE_LOW>;
+	vmmc-supply = <&reg_vsd_3v3>;
+	vqmmc-supply = <&vldo2_reg>;
 	status = "okay";
 };
 
diff --git a/arch/arm/boot/dts/imx7ulp.dtsi b/arch/arm/boot/dts/imx7ulp.dtsi
index 71b8bee..9330ba2 100644
--- a/arch/arm/boot/dts/imx7ulp.dtsi
+++ b/arch/arm/boot/dts/imx7ulp.dtsi
@@ -307,6 +307,8 @@
 				 <&clks IMX7ULP_CLK_USDHC0>;
 			clock-names ="ipg", "ahb", "per";
 			bus-width = <4>;
+			fsl,tuning-start-tap = <20>;
+			fsl,tuning-step= <2>;
 			status = "disabled";
 		};
 
-- 
1.7.9.5

